src/payments-methods/classic/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { decodeEntities } from '@wordpress/html-entities';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { PAYMENT_METHOD_NAME } from './constants';
+import { getBlocksConfiguration } from '../../utils';
+import { AmazonContent } from './payment-methods';
+
+const settings = getBlocksConfiguration( 'amazon_payments_advanced_data' );
+const defaultLabel = __( 'Amazon Pay', 'woocommerce' );
+const label = decodeEntities( settings.title ) || defaultLabel;
+
+/**
+ * Returns a React Component.
+ *
+ * @param {object} param0  RenderedComponent and props
+ * @returns {RenderedComponent}
+ */
+const AmazonComponent = ( { RenderedComponent, ...props } ) => {
+	const [ errorMessage, setErrorMessage ] = useState( '' );
+
+	useEffect( () => {
+		if ( errorMessage ) {
+			throw new Error( errorMessage );
+		}
+	}, [ errorMessage ] );
+
+	return <RenderedComponent { ...props } />;
+};
+
+/**
+ * Label component
+ *
+ * @param {object} props Props from payment API.
+ */
+const Label = ( props ) => {
+	const { PaymentMethodLabel } = props.components;
+	return <PaymentMethodLabel text={ label } />;
+};
+
+/**
+ * Amazon Pay "Classic" payment method config object.
+ */
+export const amazonPayPaymentMethod = {
+	name: PAYMENT_METHOD_NAME,
+	label: <Label />,
+	content: <AmazonComponent RenderedComponent={ AmazonContent }/>,
+	edit: <AmazonComponent RenderedComponent={ AmazonContent }/>,
+	canMakePayment: () => true,
+	ariaLabel: label,
+	supports: {
+		features: settings?.supports ?? [],
+	},
```

Comment:
didn't quite understand what you mean i'm afraid. If you are referring to the `label` variable it has a default value "Amazon Pay". If you suggest we change this default to "Proceed to Amazon" its fine by me :D

Reply:
> placeOrderButtonLabel: This is an optional label which will change the default "Place Order" button text to something else when the payment method is selected. As an example, the PayPal Standard payment method [changes the text of the button to "Proceed to PayPal"](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/e089ae17043fa525e8397d605f0f470959f2ae95/assets/js/payment-method-extensions/payment-methods/paypal/index.js#L37-L40) when it is selected as the payment method for checkout because the payment method takes the shopper offsite to PayPal for completing the payment.

From [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/docs/extensibility/payment-method-integration.md#payment-methods---registerpaymentmethod-options-). I think it make sense to do the same thing as PayPal.